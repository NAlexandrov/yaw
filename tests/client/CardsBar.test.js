import React from 'react';
import CardsBar from '../../source/client/components/CardsBar';

const cardsList = [{
  id: 1,
  cardNumber: '546925000000000',
  balance: '231310',
}];

const activeCardIndex = 1;

const onCardChange = jest.fn();

describe('<CardsBar />', () => {
  test('should render a cards bar', () => {
    const wrapper = shallow(
      <CardsBar
        cardsList={cardsList}
        activeCardIndex={activeCardIndex}
        onCardChange={onCardChange}
        isCardsEditable={false}
        isCardRemoving={false}
        deleteCard={() => ({})}
        onChangeBarMode={() => ({})} />);

    expect(wrapper).toMatchSnapshot();
  });
});
