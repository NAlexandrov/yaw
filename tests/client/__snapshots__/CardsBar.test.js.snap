// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardsBar /> should render a cards bar 1`] = `
<Styled(div)>
  <Styled(div) />
  <Styled(div)>
    <Card
      active={false}
      data={
        Object {
          "balance": "231310",
          "cardNumber": "546925000000000",
          "id": 1,
        }
      }
      isCardsEditable={false}
      key="0"
      onChangeBarMode={[Function]}
      onClick={[Function]}
    />
    <Card
      type="new"
    />
  </Styled(div)>
  <Styled(footer)>
    Yamoney Node School
  </Styled(footer)>
</Styled(div)>
`;
